src/bin/miri.rs

```diff
@@ -404,6 +354,61 @@ fn jemalloc_magic() {
     }
 }
 
+fn entry_fn(tcx: TyCtxt<'_>) -> (DefId, EntryFnType) {
+    if let Some(entry_def) = tcx.entry_fn(()) {
+        entry_def
+    } else {
+        let sym = tcx.exported_symbols(LOCAL_CRATE).iter().find_map(|(sym, _)| {
+            if sym.symbol_name_for_local_instance(tcx).name == "miri_start" {
+                Some(sym)
+            } else {
+                None
+            }
+        });
+        match sym {
+            Some(ExportedSymbol::NonGeneric(id)) => {
+                let start_def_id = id.expect_local();
+                let start_span = tcx.def_span(start_def_id);
+
+                let expected_sig = ty::Binder::dummy(tcx.mk_fn_sig(
+                    [tcx.types.isize, Ty::new_imm_ptr(tcx, Ty::new_imm_ptr(tcx, tcx.types.u8))],
+                    tcx.types.isize,
+                    false,
+                    hir::Safety::Safe,
+                    Abi::Rust,
+                ));
+
+                let correct_func_sig = check_function_signature(
+                    tcx,
+                    ObligationCause::new(start_span, start_def_id, ObligationCauseCode::Misc),
+                    *id,
+                    expected_sig,
+                )
+                .is_ok();
+
+                if correct_func_sig {
+                    (*id, EntryFnType::Start)
+                } else {
+                    tcx.dcx().fatal(
+                        "`miri_start` must have the following signature:\
+                        \nfn miri_start(argc: isize, argv: *const *const u8) -> isize",
+                    );
+                }
+            }
+            _ => {
+                tcx.dcx().fatal("Miri can only run programs that have a main function.\
```

Comment:
Please put the `\n` at the ends of these lines, rather than the beginning of the next lines.